c10/cuda/CUDACachingAllocator.h

```diff
@@ -152,15 +148,15 @@ struct TraceEntry {
       int64_t addr,
       size_t size,
       cudaStream_t stream,
-      std::shared_ptr<Context> context = nullptr)
+      std::shared_ptr<GatheredContext> context = nullptr)
       : action_(action),
         addr_(addr),
         context_(context),
```

Comment:
```suggestion
        context_(std::move(context)),
```
No reason for an unnecessary copy here.